pkg/policy/manager.go

```diff
@@ -102,13 +102,14 @@ func (m *ManagerImpl) Run(ctx context.Context, request *ExecuteRequest) (*Execut
 		}
 	}
 
-	var selector []string
-	if request.Policy.meta.subPolicy != "" {
-		selector = strings.Split(request.Policy.meta.subPolicy, "/")
+	filteredPolicy := request.Policy.Filter(request.Policy.meta.subPolicy)
+	if len(filteredPolicy.Policies.All()) == 0 && len(filteredPolicy.Checks) == 0 {
```

Comment:
isn't this checked in the .Execute function?

Reply:
The check needs to be in both places in order to maintain backwards compatibility... We need to sanitize any policies being passed into the function and in order to maintain the same error message we have to return the error from within the `manager Run()` function otherwise we would have to pass the selector to the function which we are trying to avoid.